lib/galaxy/managers/base.py

```diff
@@ -977,15 +977,17 @@ def _add_parsers(self):
         Set up, extend, or alter `orm_filter_parsers` and `fn_filter_parsers`.
         """
         # note: these are the default filters for all models
-        self.orm_filter_parsers.update({
-            # (prob.) applicable to all models
-            'id': {'op': ('in')},
-            'encoded_id': {'column': 'id', 'op': ('in'), 'val': self.parse_id_list},
-            # dates can be directly passed through the orm into a filter (no need to parse into datetime object)
-            'extension': {'op': ('eq', 'like', 'in')},
-            'create_time': {'op': ('le', 'ge', 'lt', 'gt'), 'val': self.parse_date},
-            'update_time': {'op': ('le', 'ge', 'lt', 'gt'), 'val': self.parse_date},
-        })
+        self.orm_filter_parsers.update(
+            {
+                # (prob.) applicable to all models
+                "id": {"op": ("in")},
+                "encoded_id": {"column": "id", "op": ("in"), "val": self.parse_id_list},
+                # dates can be directly passed through the orm into a filter (no need to parse into datetime object)
+                "extension": self.create_str_orm_filter,
```

Comment:
This `create_str_orm_filter` function looks generic for parsing string filters, but it doesn't handle all the string filters, like `contains` for example. I guess because is kind of tailored for this `extension` column.
Would be possible to just add a value parser (similar to `parse_id_list` and the rest) that gets the `op` (as an optional parameter) for discriminating if it should split or not?

Reply:
> This `create_str_orm_filter` function looks generic for parsing string filters, but it doesn't handle all the string filters, like `contains` for example. I guess because is kind of tailored for this `extension` column.

Yes, it is tailored to `extension` but extensible if needed.

> Would be possible to just add a value parser (similar to `parse_id_list` and the rest) that gets the `op` (as an optional parameter) for discriminating if it should split or not?

Currently the `val` field for entries in the `orm_filter_parsers` dictionary is currently called in `_parse_orm_filter()` without passing the `op` field, so that wouldn't work without modifying all the parsers.
Thinking of this though, we could allow the `val` field to be a dictionary indexed by `op`, replacing:

```python
        # parse the val from string using the 'val' parser if present (otherwise, leave as string)
        val_parser = column_map.get('val', None)
        if val_parser:
            val = val_parser(val)
```

with:

```python
        # parse the val from string using the 'val' parser if present (otherwise, leave as string)
        val_parser = column_map.get("val")
        if isinstance(val_parser, dict):
            val_parser = val_parser.get(op)
        if val_parser:
            val = val_parser(val)
```

What do you think?

Reply:
Yeah, that sounds good to me :+1:

Reply:
Rebased the PR to follow this approach!